server/artifacts/artifact_server.go

```diff
@@ -27,10 +28,15 @@ type ArtifactServer struct {
 	hydrator          hydrator.Interface
 	wfArchive         sqldb.WorkflowArchive
 	instanceIDService instanceid.Service
+	artDriverFactory  artifact.NewDriverFunc
 }
 
 func NewArtifactServer(authN auth.Gatekeeper, hydrator hydrator.Interface, wfArchive sqldb.WorkflowArchive, instanceIDService instanceid.Service) *ArtifactServer {
-	return &ArtifactServer{authN, hydrator, wfArchive, instanceIDService}
+	return newArtifactServer(authN, hydrator, wfArchive, instanceIDService, artifact.NewDriver)
+}
+
+func newArtifactServer(authN auth.Gatekeeper, hydrator hydrator.Interface, wfArchive sqldb.WorkflowArchive, instanceIDService instanceid.Service, artDriverFactory artifact.NewDriverFunc) *ArtifactServer {
+	return &ArtifactServer{authN, hydrator, wfArchive, instanceIDService, artDriverFactory}
```

Comment:
Allowing a mock driver factory to be used was the simplest way that I could think of to enable the proposed change to be tested since only the S3/GCS/OSS drivers support `Key` since they're the only ones that support outputs.